Document/0x04e-Testing-Authentication-and-Session-Management.md

```diff
@@ -250,7 +250,7 @@ The purpose of this test case is verifying logout functionality and determining
 
 Failing to destroy the server-side session is one of the most common logout functionality implementation errors. This error keeps the session or token alive, even after the user logs out of the application. An attacker who gets valid authentication information can continue to use it and hijack a user's account.
 
-Many mobile apps don't automatically log users out because it is inconvenient for customers by implementing stateless authentication. The application should still have a logout function, and it should be implemented according to best practices, destroying the access and refresh token on the client and server. Otherwise, authentication can be bypassed when the refresh token is not invalidated.
+Many mobile apps don't automatically log users out because it is inconvenient for customers by implementing stateless authentication. The application should still have a logout function, and it should be implemented according to best practices, destroying the access and refresh token on the client and server. Otherwise, authentication can be bypassed when the refresh token is not invalidated. Note that other sensitive types of information should be removed as well, in order to make sure that the application does not leak any type of information.
```

Comment:
```suggestion
Many mobile apps don't automatically log users out, for example because it is inconvenient for customers, by implementing stateless authentication. The application should still have a logout function, and it should be implemented according to best practices, destroying all locally stored tokens or session identifiers. If session information is stored on the server, it should also be destroyed by sending a logout request to that server. Not removing tokens or session identifiers can result in unauthorized access to the application in case the tokens are leaked.

Note that other sensitive types of information should be removed as well, as any information that is not properly cleared may be leaked later, for example during a device backup.
```
Sorry for the heavy rewrite. Token based authentication doesn't necessarily have a serverside component. Let me know what you think.

Reply:
The first sentence still feels a bit "weird". What about..?

`Many mobile apps don't automatically log users out by implementing stateless authentication. The reason behind is usually that it is inconvenient for customers.`